lib/pactflow/client/provider_contracts/publish.rb

```diff
@@ -1,76 +1,45 @@
-          create_branch_version_and_tags
-          render_response(create_contract)
-        end
+        attr_reader :provider_name, :provider_version_number, :branch_name, :tags, :build_url, :contract, :verification_results
 
-        def render_response(res)
-          notices = [
-            { type: 'success', text: "Successfully published provider contract for #{provider_name} version #{provider_version_number} to PactFlow"},
-          ]
-          if res.body && res.body['_links'] && res.body['_links']['pf:ui']['href']
-            notices.concat([{ text: "View the uploaded contract at #{res.body['_links']['pf:ui']['href']}" }])
+        def do_call
+          if enabled? && index_resource.assert_success!.can?(PUBLISH_RELATION)
+            publish_provider_contracts
+            PactBroker::Client::CommandResult.new(success?, message)
+          else
+            PublishTheOldWay.call(params, options, pact_broker_client_options)
           end
-          notices.concat(next_steps)
-          PactBroker::Client::CommandResult.new(true, PactBroker::Client::ColorizeNotices.call(notices.collect do |n|
-                                                                                                 OpenStruct.new(n)
-                                                                                               end).join("\n"))
-        end
-
-        def next_steps
-          [
-            { type: 'prompt', text: 'Next steps:' },
-            { type: 'prompt',
-              text: '  * Check your application is safe to deploy - https://docs.pact.io/can_i_deploy' },
-            { text: "       pact-broker can-i-deploy --pacticipant #{provider_name} --version #{provider_version_number} --to-environment <your environment name>" },
-            { type: 'prompt',
-              text: '  * Record deployment or release to specified environment (choose one) - https://docs.pact.io/go/record-deployment' },
-            { text: "       pact-broker record-deployment --pacticipant #{provider_name} --version #{provider_version_number} --environment <your environment name>" },
-            { text: "       pact-broker record-release --pacticipant #{provider_name} --version #{provider_version_number} --environment <your environment name>" }
-          ]
         end
 
-        def create_branch_version_and_tags
-          if branch_name || tags.any?
-            pacticipant_version_params = {
-              pacticipant_name: provider_name,
-              version_number: provider_version_number,
-              branch_name: branch_name,
-              tags: tags
-            }
-            result = PactBroker::Client::Versions::Create.call(pacticipant_version_params, options, pact_broker_client_options)
-            if !result.success
-              raise PactBroker::Client::Error.new(result.message)
-            end
-          end
+        def enabled?
+          ENV.fetch("PACT_BROKER_FEATURES", "").include?("publish_provider_contracts_all_in_one")
```

Comment:
Would the user set this? 

will it say be set in the pact-cli and pact-ruby-standalone packages for the pactflow command, or are we expecting this to be used in testing for a bit and then we use the all in one?

Reply:
why is the old way of publishing hanging around, whats the craic in general. is that meant to be phased out over time

Reply:
The feature toggle is so we can use it internally for a few days before putting it out for public use. It's what I do for all our new commands. The code will be included in the docker images and ruby standalone as soon as I release it (today).

The old way is required because this is a brand new API, and our onprem customers will be using previous versions of Pactflow for who-knows-how-long. Some only update once a year or so.

Reply:
Ok fabulous, that sounds like a really well considered approach, thanks for clarifying @bethesque